tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -369,11 +385,12 @@ def _node_leadership_evacuated(self, topic_names: list[str],
                 continue
 
             assert len(partitions) == p_per_topic
-            for p in partitions:
-                if p.leader == node_id:
-                    self.logger.info(
-                        f"partition {tn}/{p.id} still on node {node_id}")
-                    any_incomplete = True
+            remaining = sum([1 for p in partitions if p.leader == node_id])
```

Comment:
Not that it matters much here but you don't need the temporary list/`[`. Can just use a generator expression.